### DISCUSSION TITLE
SwitchNetwork for WalletConnect

### DISCUSSION THREAD

**Discussion by naz3eh** [wevm/viem#753] [Question] [answered]
I am trying to switch networks using the `walletClient.switchChain()`  method after connecting the wallet using WalletConnect. It does not give me an error but it is not popping up the confirmation in the wallet. 

Some folks have reported a similar issue with rainbowkit [here](https://github.com/rainbow-me/rainbowkit/issues/1307#issuecomment-1601142395) and as it uses wagmi and thereby viem bts, it might be an issue with the viem.

Some code links: 
- Configuring WalletConnect and creating walletClient: https://github.com/Nazeeh21/comp-kit/blob/1bab045ea5f85a7f86ddf49ec615fad6712a69b6/package/src/components/Wallets/walletConnectWallet.tsx#L44
- SwitchNetwork code: https://github.com/Nazeeh21/comp-kit/blob/1bab045ea5f85a7f86ddf49ec615fad6712a69b6/package/src/hooks/useSwitchChain/index.tsx#L43

**Comment by jxom** [comment, marked as answer]
WalletConnect v2 does not prompt the Wallet to switch networks. When we instantiate a WalletConnect v2 Provider, you provide it with a set of chains that you support, and then when the user connects, it will prompt them to authorize ALL supported chains. This means that no prompt will appear in the Wallet when you call `wallet_switchEthereumChain`, because the chain has already been authorized. Instead, WalletConnect v2's Provider switches chain context when this method is called, and you need to handle active chain state accordingly.

**Comment by naz3eh** [comment]
Gotcha! Tysm for clearing up my confusion and helping me out. :)

**Comment by jingxuan98** [comment]
I am using metamask Mobile Wallet, but when call the switchNetwork() from 'wagmi', it switches the walletClient and signer but no switchChain pop up in metamask mobile wallet app

So its remaining at the same network page in metamask wallet and when initiate a contract interaction, theres no pop up for me to sign as well in metamask mobile, while everything works fine if i manually switch to the correct network in metamask mobile

How can i handle this? @jxom

**Reply by AdiDubey0415** [reply]
Same issue, can someone please help here?

**Reply by AdiDubey0415** [reply]
@jxom can you please help here?
